### DISCUSSION TITLE
Bring compatibility to OrbStack

### DISCUSSION THREAD

**Discussion by jonathanlaniado** [louislam/dockge#204] [Feature Request]
### ⚠️ Please verify that this bug has NOT been reported before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [x] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### Description

Dockge is unable to connect to the Docker daemon on Orbstack. Despite mounting the Docker socket `/var/run/docker.sock` from the host into the container, Dockge tries to directly access the Docker socket on the host OS at path `/Users/(redacted)/.orbstack/run/docker.sock`, which is not accessible from within the Docker container.

### 👟 Reproduction steps

1. Use Orbstack instead of Docker Desktop on macOS.
2. Set up Dockge's compose file with this volume:
```
volumes:
  - /var/run/docker.sock:/var/run/docker.sock
  - ...
```
3. Start Dockge.

### 👀 Expected behavior

Dockge should be able to successfully communicate with the Docker daemon using the Docker socket mounted into the container at `/var/run/docker.sock`.

### 😓 Actual Behavior

However, Dockge appears to be trying to communicate with Docker daemon via the host's filesystem directly at `/Users/(redacted)/.orbstack/run/docker.sock`. This operation ends up in failure as a Docker container cannot directly access the file system of its host, resulting in the error:

```Cannot connect to the Docker daemon at unix:///Users/(redacted)/.orbstack/run/docker.sock. Is the docker daemon running?```

### Dockge Version

1.2

### 💻 Operating System and Arch

macOS Sonoma 14.1.1 (23B81)

### 🌐 Browser

Arc 1.17.2 (43249) Chromium Engine 119.0.6045.159

### 🐋 Docker Version

24.0.6

### 📝 Relevant log output

```shell
dockge  | 2023-11-28T15:21:03Z [SERVER] INFO: Welcome to dockge!
dockge  | 2023-11-28T15:21:03Z [SERVER] INFO: NODE_ENV: production
dockge  | 2023-11-28T15:21:03Z [SERVER] INFO: Data Dir: ./data/
dockge  | 2023-11-28T15:21:03Z [SERVER] INFO: Server Type: HTTP
dockge  | 2023-11-28T15:21:03Z [DB] INFO: Database Type: sqlite
dockge  | 2023-11-28T15:21:03Z [SERVER] INFO: Connected to the database
dockge  | 2023-11-28T15:21:03Z [SERVER] INFO: Listening on 5001
dockge  | 2023-11-28T15:21:03Z [SERVER] INFO: Socket connected!
dockge  | 2023-11-28T15:21:03Z [AUTH] INFO: Login by token. IP=192.168.228.1
dockge  | 2023-11-28T15:21:03Z [AUTH] INFO: Username from JWT: jonathan
dockge  | Cannot connect to the Docker daemon at unix:///Users/(redacted)/.orbstack/run/docker.sock. Is the docker daemon running?
dockge  | 2023-11-28T15:21:03Z [SERVER] ERROR: Error: Command failed: docker compose ls --all --format json
dockge  | Cannot connect to the Docker daemon at unix:///Users/(redacted)/.orbstack/run/docker.sock. Is the docker daemon running?
dockge  | 
dockge  |     at __node_internal_genericNodeError (node:internal/errors:865:15)
dockge  |     at checkExecSyncError (node:child_process:885:11)
dockge  |     at Object.execSync (node:child_process:957:15)
dockge  |     at Function.getStackList (/app/backend/stack.ts:225:32)
dockge  |     at DockgeServer.sendStackList (/app/backend/dockge-server.ts:497:43)
dockge  |     at DockgeServer.afterLogin (/app/backend/dockge-server.ts:248:18)
dockge  |     at Socket.<anonymous> (/app/backend/socket-handlers/main-socket-handler.ts:76:34) {
dockge  |   status: 1,
dockge  |   signal: null,
dockge  |   output: [
dockge  |     null,
dockge  |     <Buffer >,
dockge  |     <Buffer 43 61 6e 6e 6f 74 20 63 6f 6e 6e 65 63 74 20 74 6f 20 74 68 65 20 44 6f 63 6b 65 72 20 64 61 65 6d 6f 6e 20 61 74 20 75 6e 69 78 3a 2f 2f 2f 55 73 65 ... 69 more bytes>
dockge  |   ],
dockge  |   pid: 48,
dockge  |   stdout: <Buffer >,
dockge  |   stderr: <Buffer 43 61 6e 6e 6f 74 20 63 6f 6e 6e 65 63 74 20 74 6f 20 74 68 65 20 44 6f 63 6b 65 72 20 64 61 65 6d 6f 6e 20 61 74 20 75 6e 69 78 3a 2f 2f 2f 55 73 65 ... 69 more bytes>
dockge  | }
```

**Comment by jonathanlaniado** [comment]
Hi @louislam, do you plan on fixing this? 😊

**Reply by louislam** [reply]
I don't own Mac machine. 

But I feel that is more like a config issue on your side. You should ask OrbStack's community.

**Reply by jonathanlaniado** [reply]
I wonder if there’s a minor tweak you can do to ensure Dockge uses the correct socket path. I am happy to test this out from a Mac perspective, as this image and Uptime Kuma are such fantastic pieces of software.

I have the same volumes set up with Portainer and it works flawlessly:
```
volumes:
  - /var/run/docker.sock:/var/run/docker.sock
  - …
```

Regarding OrbStack, there's no configuration on my end. Orbstack is pretty much set up right out of the box. Here’s their documentation on this:
>OrbStack creates a Docker context called orbstack. This context is automatically used when you run commands from the terminal.
>
>If you have admin access, OrbStack will also create a symlink at /var/run/docker.sock to improve compatibility with some third-party tools.

Really hope this is an easy fix and you can help with this! 👍

**Reply by louislam** [reply]
Actually Dockge don't communicate with the Docker API via docker.sock, Dockge internally is also calling commands such as "docker compose up/down" directly. 

I think you should go into Dockge's bash and find out what happened.

```
docker compose exec dockge bash
```

You can try these commands inside Dockge's bash:
```
docker version
docker container ls
docker compose ls
```
